pubsub/redis.go

```diff
@@ -65,10 +75,79 @@ func (s *RedisSubscriber) Start() error {
 }
 
 func (s *RedisSubscriber) listen() error {
-	c, err := redis.DialURL(s.url)
 
-	if err != nil {
-		return err
+	var c redis.Conn
+	var err error
+
+	if s.sentinelsEnabled {
+
+		s.log.Debugf("Redis sentinel enabled")
+		s.log.Debugf("Redis sentinel parameters:  sentinels: %s,  masterName: %s", s.sentinels, s.masterName)
+		sentinels := strings.Split(s.sentinels, ",")
+		sntnl := &sentinel.Sentinel{
+			Addrs:      sentinels,
+			MasterName: s.masterName,
+			Dial: func(addr string) (redis.Conn, error) {
+				timeout := 500 * time.Millisecond
+
+				c, err := redis.Dial(
+					"tcp",
+					addr,
+					redis.DialConnectTimeout(timeout),
+					redis.DialReadTimeout(timeout),
+					redis.DialReadTimeout(timeout),
+				)
+				if err != nil {
+					s.log.Debugf("Failed to connect to sentinel %s", addr)
+					return nil, err
+				}
+				s.log.Debugf("Successfully connected to sentinel %s", addr)
+				return c, nil
+			},
+		}
+
+		defer sntnl.Close()
+
+		pool := &redis.Pool{
```

Comment:
Do we need a pool here? We're only using a single connection.

We only need a role check right after the `Dial`.